### ISSUE TITLE
Out of combat health regeneration seems off

### ISSUE THREAD

**Issue by Apathyxia** [chromiecraft/chromiecraft#6057]
### What client do you play on?

enGB

### Faction

Both

### Content Phase:

Generic

### Current Behaviour

Base health regeneration is roughly 20-25 health per tick out of combat on all levels (tested on level 70 druid, 30 hunter and 1 druid), when there is no extra spirit from gear. 

That is:
Level 70 night elf druid regens 25 health per tick
Level 30 troll hunter gained 22 health per tick
Level 1 tauren druid gained 23 health per tick

![image](https://github.com/chromiecraft/chromiecraft/assets/107246426/c9bccdc7-dbb1-4958-93aa-0b837c08a4c3)

![image](https://github.com/chromiecraft/chromiecraft/assets/107246426/a1519549-5779-45ec-99ac-55a9a1dd4726)

![image](https://github.com/chromiecraft/chromiecraft/assets/107246426/88640e2d-4e39-4f69-b248-7d193148acd9)

The effects of base health regeneration are surely a plus for a low level characters, but it makes no sense my 70 druid basically regens roughly the same amount as a level 1 druid. 

Even looking at tooltips and multiplying by 2 to consider a health regen tick, the values still don't add up to what they actually are, or what spirit based regeneration calculations yield (see expected blizzlike behaviour).

### Expected Blizzlike Behaviour

Here is some regeneration calculations based on spirit: https://wowwiki-archive.fandom.com/wiki/Health_regeneration

Using calculations from above link:
Level 70 druid with 140 spirit should have 140 * 0.09 + 6.5 health per tick = 19.1 health per tick.
Level 30 hunter with 56 spirit should have (56 * 0.25 + 6)*1.1 health per tick = 22 health per tick **(This is correct)**
Level 1 druid with 24 spirit should have 24 * 0.09 + 6.5 health per tick = 8.66 health per tick.

### Source

https://wowwiki-archive.fandom.com/wiki/Health_regeneration

### Steps to reproduce the problem

1. Make a character (anything, really)
2. Take some damage
3. Observe health regeneration (and check the spirit tooltip)

### Extra Notes

I am not entirely sure if there's some hidden regeneration working as well. Level 8 priest apparently also regens 25 health per tick with 39 spirit. 

Level 62 warrior with 62 spirit regens 74 health per tick, when calculating the value yields 55.6. 

In general, I think health regeneration values are incorrectly calculated.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/86bfef8874883a1726966326368918567c5531cd

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-BreakingNewsOverride](https://github.com/AnchyDev/BreakingNewsOverride)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-zone-difficulty](https://github.com/azerothcore/mod-zone-difficulty)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-event-scripts](https://github.com/55Honey/Acore_eventScripts)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

**Comment by Salehyn** [comment]
I have a few notes as i've been recently testing this:
1) Low level characters do get a complementary regen on 3.3.5a which slowly weans off level 1-15 (don't ask me about the formula), so their high regen itself is normal, although the numbers are hella random on CC
2) the spirit tooltip is visual bugged, it says per second when it really means 'per tick' (plus the numbers are fucked by rounding). The formula on wiki is for per tick also, so every 2second... so i think your calculations are off. 140*0.09+6.5 makes 19.1 per tick.
3) No matter how you calculate the numbers are never gonna match, sometimes it's more, sometimes it's less, i've quit trying to make sense of it.
4) I found a bunch of other anomalies when testing regen at different brackets the numbers just go up and down and up again these are single digit differences that aren't worth my mental health so i've let it go.

TL;DR i do think there are lots of issues with this, but double check the calculations

**Comment by Apathyxia** [comment]
> I have a few notes as i've been recently testing this:
> 
> 1. Low level characters do get a complementary regen on 3.3.5a which slowly weans off level 1-15 (don't ask me about the formula), so their high regen itself is normal, although the numbers are hella random on CC
> 2. the spirit tooltip is visual bugged, it says per second when it really means 'per tick' (plus the numbers are fucked by rounding). The formula on wiki is for per tick also, so every 2second... so i think your calculations are off. 140*0.09+6.5 makes 19.1 per tick.
> 3. No matter how you calculate the numbers are never gonna match, sometimes it's more, sometimes it's less, i've quit trying to make sense of it.
> 4. I found a bunch of other anomalies when testing regen at different brackets the numbers just go up and down and up again these are single digit differences that aren't worth my mental health so i've let it go.
> 
> TL;DR i do think there are lots of issues with this, but double check the calculations

Yeah, you are actually right - the calculations are wrong. I had calculated them initially per second, and realised them to be per tick. Forgot to switch the latter part where I multiplied them by 2. Fixed now.

**Comment by Salehyn** [comment]
Testing done by Kerrik on wotlk classic shows that the druid spirit coefficient is in fact higher than intended on CC.
Expected calculated value 9% (blizzlike) / calculated value 13% (cc) at lvl 70.
Test1 done with a naked lvl 70 druid with 135 spirit, regen 18-19 per tick (blizzlike)
Test2 done with a geared lvl 70 druid with 241 spirit, regen 28-29 per tick (blizzlike)
Test3 done with a naked lvl 70 druid with 135 spirit, regen 24 per tick (cc)
Test4 done with a geared lvl 70 druid with 241 spirit, regen 37 per tick (cc)
Base value =6.5 per tick
Other level ranges and classes still under testing.

**Comment by Salehyn** [comment]
Looks like classic values might be incorrect here. Buggy blizzard.
